### PR TITLE
feat(autoware_ad_api_msgs): add response status

### PR DESCRIPTION
**Pull request by isamu-takagi**
Signed-off-by: Takagi, Isamu <isamu.takagi@tier4.jp>

## Description

Add the response status message and update the readme. The links depend on:
- https://github.com/autowarefoundation/autoware-documentation/pull/115
- https://github.com/autowarefoundation/autoware-documentation/pull/136

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/autoware_ad_api_msgs/README.md [hunk 1]
```diff
@@ -1,17 +1,25 @@
 # autoware_ad_api_msgs
 
-## InterfaceVersion
+## ResponseStatus
+
+This message is a response status commonly used in the service type API. For details, see [the response status][docs-response-status].
```
Comment (kenji-miyake) [inline common/autoware_ad_api_msgs/README.md:5]: Currently, it's linked not to your PR (https://github.com/autowarefoundation/autoware-documentation/pull/136) but to the old documentation.

For better maintainability, I think it's good to manage only high-level overviews in `autoware-documentation` and manage details alongside the implementations.

So, I mean, I recommend putting the details in this msg package and linking there from `autoware-documentation`.

@LalithVipulananthan What do you think about this?

File: common/autoware_ad_api_msgs/common/msg/ResponseStatus.msg [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# constants for level
+uint16 UNKNOWN=0
+uint16 SUCCESS=1
+uint16 WARNING=2
+uint16 ERROR=3
+
+# constants for code
+uint16 INTERNAL_SERVICE_UNREADY=101
+uint16 INTERNAL_SERVICE_TIMEOUT=102
```
Comment (kenji-miyake) [inline common/autoware_ad_api_msgs/common/msg/ResponseStatus.msg:9]: I think the prefix `INTERNAL_SERVICE_` can be removed.
Also, could you explain why you start from `101`?

```suggestion
uint16 UNREADY=101
uint16 TIMEOUT=102
```

File: common/autoware_ad_api_msgs/common/msg/ResponseStatus.msg [hunk 3]
```diff
@@ -0,0 +1,14 @@
+# constants for level
+uint16 UNKNOWN=0
+uint16 SUCCESS=1
+uint16 WARNING=2
+uint16 ERROR=3
```
Comment (kenji-miyake) [inline common/autoware_ad_api_msgs/common/msg/ResponseStatus.msg:5]: I guess we can remove `UNKNOWN`. For what use cases did you add this?

```suggestion
uint16 SUCCESS=0
uint16 WARNING=1
uint16 ERROR=2
```

File: common/autoware_ad_api_msgs/README.md [hunk 4]
```diff
@@ -1,17 +1,25 @@
...
+Each API defines its own error code and assigns a level to them. For example, define 12345 as the code to indicate that the parameter is invalid.
+The level of this code is an error if the process cannot be continued, or a warning if the process can be continued by using the default value or something.
```
Comment (kenji-miyake) [inline common/autoware_ad_api_msgs/README.md:7]: > a warning if the process can be continued by using the default value or something.

Could you elaborate on this part? I'd like to know a bit more concrete examples.
I couldn't understand what is `the default value or something` and when we should use them.

File: common/autoware_ad_api_msgs/common/msg/ResponseStatus.msg [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+
+# constants for code
+uint16 INTERNAL_SERVICE_UNREADY=101
+uint16 INTERNAL_SERVICE_TIMEOUT=102
+
+# variables
+uint16 level
+uint16 code
```
Comment (kenji-miyake) [inline common/autoware_ad_api_msgs/common/msg/ResponseStatus.msg:13]: Why did you make this a number? I guess we can use a string instead.

**Comment by isamu-takagi** [comment]
I discussed with @kenji-miyake and @mitsudome-r.
I decided to use the boolean value instead of the error level because this is simpler and easier to understand. https://github.com/autowarefoundation/autoware.universe/pull/1042/commits/26e6e5a3153c0a3ee2d01248256c29c693c85f4b

**Pull request merged**